sql/parser/builtins.go

```diff
@@ -761,3 +774,12 @@ func datumToRawString(datum Datum) (string, error) {
 
 	return "", fmt.Errorf("argument type unsupported: %s", datum.Type())
 }
+
+func round(x float64, n int64) (Datum, error) {
+	if n < 0 || n > 50 {
```

Comment:
Yes. Return an error in round() for negative `ndigits` and the original number for `ndigits > 500`.